base/operators.jl

```diff
@@ -61,6 +61,13 @@ const ≥ = >=
 .>=(x,y) = y .<= x
 const .≥ = .>=
 
+for op in (<, >, <=, >=, (==))
+    # to fix ambiguities
+    @eval promote_op{R<:Number,S<:Number}(::$(typeof(op)), ::Type{R}, ::Type{S}) =
+        ($(Expr(:meta, :pure)); Bool)
+    @eval promote_op{Op<:$(typeof(op))}(::Op, ::Any, ::Any) = ($(Expr(:meta, :pure)); Bool)
```

Comment:
I think I do, because the general definition for `Number` applies to all operations, while this one applies only to one operation, for for all input types. So there's still an ambiguity.